frontend/src/containers/Calendar/Calendar.test.js

```diff
@@ -1,18 +1,82 @@
 import dayjs from 'dayjs';
+import {calcAvailTimeSlot} from './Index';
 
-function onCalChangeHandler(e) {
-  const newDate = dayjs().format('2020-03-18');
-  if (e === 'Prev') {
-    newDate = dayjs(newDate).subtract(7, 'day');
-  } else {
-    newDate = dayjs(newDate).add(7, 'day');
+const selectDateEmpty = "2020-05-03";
+const selectDateFill = "2020-04-28";
+const jobObjEmpty = {
+    "2020-05-03": {},
+    "2020-05-04": {},
+    "2020-05-05": {},
+    "2020-05-06": {},
+    "2020-05-07": {},
+    "2020-05-08": {},
+    "2020-05-09": {}
   }
-  return newDate.format('YYYY-MM-D');
-}
-
-test('date changer', () => {
-  // Next test
-  expect(onCalChangeHandler('Next')).toBe('2020-03-25');
-  // Prev test
-  expect(onCalChangeHandler('Prev')).toBe('2020-03-11');
+
+const jobObjFound = {
+    "2020-04-26": {},
+    "2020-04-27": {},
+    "2020-04-28": {
+      "7": {
+        "jobId": 4,
+        "clientId": 1
+      },
+      "15": {
+        "jobId": 7,
+        "clientId": 1
+      }
+    },
+    "2020-04-29": {},
+    "2020-04-30": {},
+    "2020-05-01": {},
+    "2020-05-02": {}
+  }
+
+
+test('GenerateTimeslotArr', () => {
+  let arr = calcAvailTimeSlot(jobObjFound,selectDateFill);
+
+  const expected = [1, 2, 3, 4, 5, 6];
+  expect([4, 1, 6, 7, 3, 5, 7, 5, 4, 6]).not.toEqual(
+    expect.arrayContaining(expected),
+  );
+
+  //return type is array 
+  expect(arr).toEqual(expect.any(Array))
+  //return obj in array
+  expect(arr).toEqual(
+    expect.arrayContaining([
+      expect.objectContaining({
```

Comment:
Should do checks for expect.toMatchObject and compare to exact expected output. A containing check is not enough as what if some new change adds more than what it's expecting? This test would not fail and you now have unexpected behaviour